pandas/core/dtypes/inference.py

```diff
@@ -282,12 +285,20 @@ def is_list_like(obj):
     >>> is_list_like(np.array(2)))
     False
     """
-
-    return (isinstance(obj, compat.Iterable) and
-            # we do not count strings/unicode/bytes as list-like
-            not isinstance(obj, string_and_binary_types) and
-            # exclude zero-dimensional numpy arrays, effectively scalars
-            not (isinstance(obj, np.ndarray) and obj.ndim == 0))
+    if strict is None and isinstance(obj, set):
```

Comment:
I think the `isinstance` checks should be against something like `collections.abc.Set` to catch more exotic `set` variations:
```python
In [2]: x = list('aabbc')

In [3]: s1 = set(x)

In [4]: s2 = frozenset(x)

In [5]: isinstance(s2, set)
Out[5]: False

In [6]: isinstance(s2, collections.abc.Set)
Out[6]: True

In [7]: isinstance(s1, collections.abc.Set)
Out[7]: True
```